### PR TITLE
BUG: passing a non-dict mapping to pd.concat raises a TypeError

### LINKED ISSUE #32863: BUG: passing a non-dict mapping to pd.concat raises a TypeError
#### Code Sample, a copy-pastable example if possible

```python
class MyMapping(Mapping):
    def __init__(self, content=None):
        if content:
            self._content = content
        else:
            self._content = {}
    def __getitem__(self, key):
        return self._content[key]
    def __iter__(self):
        return iter(self._content)
    def __len__(self):
        return len(self._content)   

objs = MyMapping({"a": pd.DataFrame({"_0": [0.], "_1": [0.]}), "b": pd.DataFrame({"_0": [0.], "_1": [0.]})}
concat_objs = pd.concat(objs, axis=1)
```
#### Problem description

The above code raises a TypeError with the message "cannot concatenate object of type '<class 'str'>'; only Series and DataFrame objs are valid".

As the documentation explicitly specifies that the argument `objs` can be a mapping of Series or DataFrame objects, the expected behaviour is to just proceed.

#### Expected Output

The concatenated data frame.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.3.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_GB.UTF-8
pandas           : 0.25.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 19.3.1
setuptools       : 41.0.0
Cython           : None
pytest           : 5.3.2
hypothesis       : None
sphinx           : 1.8.5
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.3.1
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : None
bottleneck       : 1.2.1
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.3.1
matplotlib       : 3.1.3
numexpr          : 2.7.1
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : 3.6.1
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
</details>

### LINKED ISSUE #32954: Fixture `non_mapping_dict_subclass` is wrongly named
#### Problem description

[this should explain **why** the current behaviour is a problem and why the expected output is a better solution.]

In `conftest.py`, a fixture is defined with the name `non_mapping_dict_subclass`, which returns a minimal class implementing the `collections.abc.Mapping` base class by minimally wrapping a `dict`.

Obviously, the intended name was `non_dict_mapping_subclass`, which also is the name of the class returned (casing apart).

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 1d13390e26bc7b1f5fb597bc5df1591f5d160e1f
python           : 3.6.10.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.3.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_GB.UTF-8
pandas           : 1.0.3
numpy            : 1.18.2
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.0.0.post20200309
Cython           : 0.29.15
pytest           : 5.4.1
hypothesis       : 5.7.0
sphinx           : 2.4.4
blosc            : 1.8.3
feather          : None
xlsxwriter       : 1.2.8
lxml.etree       : 4.3.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.13.0
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : 1.2.1
fastparquet      : 0.3.3
gcsfs            : None
lxml.etree       : 4.3.1
matplotlib       : 3.2.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.1
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pytest           : 5.4.1
pyxlsb           : None
s3fs             : 0.4.0
scipy            : 1.4.1
sqlalchemy       : 1.3.15
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : 0.15.0
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.2.8
numba            : 0.48.0
</details>

### PR DESCRIPTION
**Pull request by pbourguignon**
closes #32954 
closes #32863 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.1.0.rst [hunk 1]
```diff
@@ -412,6 +414,7 @@ Other
 - Fixed bug in :func:`pandas.testing.assert_series_equal` where dtypes were checked for ``Interval`` and ``ExtensionArray`` operands when ``check_dtype`` was ``False`` (:issue:`32747`)
 - Bug in :meth:`Series.map` not raising on invalid ``na_action`` (:issue:`32815`)
 - Bug in :meth:`DataFrame.__dir__` caused a segfault when using unicode surrogates in a column name (:issue:`25509`)
+- Fixture :func:`non_mapping_dict_subclass` is now more appropriately called :func:`non_dict_mapping_subclass` (:issue:`32954`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst]: can you remove this; we don't add testing things to the release notes

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -385,8 +385,10 @@ Reshaping
 - :meth:`DataFrame.replace` and :meth:`Series.replace` will raise a ``TypeError`` if ``to_replace`` is not an expected type. Previously the ``replace`` would fail silently (:issue:`18634`)
 - Bug in :meth:`DataFrame.apply` where callback was called with :class:`Series` parameter even though ``raw=True`` requested. (:issue:`32423`)
 - Bug in :meth:`DataFrame.pivot_table` losing timezone information when creating a :class:`MultiIndex` level from a column with timezone-aware dtype (:issue:`32558`)
+- Bug in :meth:``pd.concat`` where when passing a non-dict mapping as ``objs`` would raise a TypeError (:issue:`32863`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.1.0.rst]: use ```:meth:`concat` ``` instead

put double back-ticks around ```TypeError```

**Review by jreback — CHANGES_REQUESTED** [review]
small comments, ping on green.

File: doc/source/whatsnew/v1.1.0.rst (hunk 1 above)
Comment (pbourguignon) [reply doc/source/whatsnew/v1.1.0.rst]: Done

File: doc/source/whatsnew/v1.1.0.rst (hunk 2 above)
Comment (pbourguignon) [reply doc/source/whatsnew/v1.1.0.rst]: Done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @pbourguignon

**Pull request merged**